File: SmartHomeTEC_API/Models/ChamberAssociation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SmartHomeTEC_API.Models
{
    [Table("ChamberAssociation")]
    public class ChamberAssociation
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AssociationID { get; set; }

        [Required]
        [RegularExpression(@"^\d{4}-\d{2}-\d{2}$", ErrorMessage = "AssociationStartDate debe tener el formato 'YYYY-MM-DD'.")]
        public required string AssociationStartDate { get; set; } // Formato: YYYY-MM-DD

        [RegularExpression(@"^\d{4}-\d{2}-\d{2}$", ErrorMessage = "WarrantyEndDate debe tener el formato 'YYYY-MM-DD'.")]
        public string? WarrantyEndDate { get; set; } // Formato: YYYY-MM-DD

        [Required]
        [ForeignKey("Chamber")]
        public required int ChamberID { get; set; }

        [Required]
        [ForeignKey("AssignedDevice")]
        public required int AssignedID { get; set; }

        // Propiedades de Navegación:
        [JsonIgnore]
        public Chamber? Chamber { get; set; }

        [JsonIgnore]
        public AssignedDevice? AssignedDevice { get; set; }
    }
}

File: smart_home_tec_app/lib/JSONmodels/chamber_association.dart
import 'dart:convert';

ChamberAssociation chamberAssociationFromJson(String str) => ChamberAssociation.fromJson(json.decode(str));

String chamberAssociationToJson(ChamberAssociation data) => json.encode(data.toJson());

class ChamberAssociation {
  final int associationID;
  final String associationStartDate;
  final String warrantyEndDate;
  final int chamberID;
  final int assignedID;

  ChamberAssociation({
    required this.associationID,
    required this.associationStartDate,
    required this.warrantyEndDate,
    required this.chamberID,
    required this.assignedID,
  });

  factory ChamberAssociation.fromJson(Map<String, dynamic> json) => ChamberAssociation(
        associationID: json["associationID"],
        associationStartDate: json["associationStartDate"],
        warrantyEndDate: json["warrantyEndDate"],
        chamberID: json["chamberID"],
        assignedID: json["assignedID"],
      );

  Map<String, dynamic> toJson() => {
        "associationID": associationID,
        "associationStartDate": associationStartDate,
        "warrantyEndDate": warrantyEndDate,
        "chamberID": chamberID,
        "assignedID": assignedID,
      };
}
